google/cloud/storage/notification.py

```diff
@@ -253,12 +253,18 @@ def create(self, client=None, timeout=_DEFAULT_TIMEOUT, retry=None):
         :type retry: google.api_core.retry.Retry or google.cloud.storage.retry.ConditionalRetryPolicy
         :param retry:
             (Optional) How to retry the RPC. See: :ref:`configuring_retries`
+
+        :raises ValueError: if the notification already exists.
+        :raises ValueError: if the topic name is not set.
         """
         if self.notification_id is not None:
             raise ValueError(
                 "Notification already exists w/ id: {}".format(self.notification_id)
             )
 
+        if self.topic_name is None:
```

Comment:
There's two parts to answer your question.

If the topic is not included, the server returns a 400 with error message `"Invalid Google Cloud Pub/Sub topic. It should look like '//pubsub.googleapis.com/projects/*/topics/*.'"`

In our client, on [line 270](https://github.com/googleapis/python-storage/blob/b32d2ccabef21fa3dbdfba34acde85618a785d48/google/cloud/storage/notification.py#L270), we preprocess the request body `topic` through string interpolation. So topic becomes `"//pubsub.googleapis.com/projects/<project-identifier>/topics/None"` if  `topic_name=None`. Thus we get a 400 for another reason `"Cloud Pub/Sub topic '//pubsub.googleapis.com/projects/project/topics/None' not found, ..."`

To avoid overdoing client-side validation, I could instead change the logic around preparing/formatting the topic.

Reply:
Ah okay-- yeah I think in this case since we are constructing the topic client-side using `topic_name`, that's a good reason to just do the validation ourselves. It would be confusing if users thought that they had to supply the whole path in the error message you show as `topic_name`. Let's just try to make as clear a message as possible for this.